nodestream_akamai/akamai_utils/property_client.py

```diff
@@ -271,48 +274,124 @@ def search_akamai_rule_tree_for_origins(self, rule_tree) -> List[Origin]:
         ]
         for behavior in behaviors:
             if behavior.get("name") == "origin":
-                origin_options = behavior["options"]
-                hostname = origin_options.get("hostname")
-                if hostname is not None:
-                    origins.append(Origin(name=hostname))
+                if "hostname" in behavior["options"].keys():
+                    origin_host = behavior["options"]["hostname"]
+                elif "netStorage" in behavior["options"].keys():
+                    origin_host = behavior["options"]["netStorage"][
+                        "downloadDomainName"
+                    ]
+                elif "mslorigin" in behavior["options"].keys():
+                    origin_host = behavior["options"]["mslorigin"]
+
+                origins.append(Origin(name=origin_host))
 
         # TODO: Maybe find a less terrible way to get unique list of origins - check as you go?
         return list(set(origins))
 
     def collate_origins_with_criteria(self, rules):

```

Comment:
This method is pretty verbose to follow, can we break the logic into more methods/functions?

Reply:
The only thing I can think to do to make it shorter is to move chunks of code into their own functions, but I don't believe that helps in making it easier to understand. Quite the opposite, in fact. It is a complex function, though, so I could help readers by adding more comments describing its operation, if that is of interest?